### DISCUSSION TITLE
How to customise event handlers of slate

### DISCUSSION THREAD

**Discussion by shanthi-techie** [react-page/react-page#1169] [Q&A]
How can i customise the slate event(onchange when key='enter') to add 2 new lines when enter key is pressed.

**Comment by macrozone** [comment]
> How can i customise the slate event(onchange when key='enter') to add 2 new lines when enter key is pressed.

enter key makes a new paragraph and shift+enter makes a line break. i highly recommend to stick with that and use proper formatting.

otherwise best look how this could be done in slate and then we can check whether and how we could implement it

**Reply by shanthi-techie** [reply]
![image](https://user-images.githubusercontent.com/66308775/156544504-dda59fdf-8f0b-46fa-badb-381478ee1383.png)

Use case :
I'm adding a form component with a label and input in the slate, This component occupies two lines of space. The cursor is still in the first line and moves the form component to next line when enter is pressed.  Also after adding a form component cursor point gets lost, i've to click on end key multiple time to get the key
